src/sentry/tasks/base.py

```diff
@@ -129,6 +134,17 @@ def _wrapped(*args, **kwargs):
 
             return result
 
+        # If the split task router is configured for the task, always use queues defined
+        # in the split task configuration
+        if name in settings.CELERY_SPLIT_QUEUE_TASK_ROUTES:
+            # TODO: remove this option once rolled out
+            if options.get("split_queue_task_router.enable"):
```

Comment:
this doesn't work -- this gets called at the module scope and will never update throughout the lifecycle of the request

the options call needs to happen at runtime

this is currently triggering a bunch of warnings in development (and spamming logs in production!):

```
/Users/asottile/workspace/sentry/.venv/lib/python3.12/site-packages/django/db/backends/utils.py:98: RuntimeWarning: Accessing the database during app initialization is discouraged. To fix this warning, avoid executing queries in AppConfig.ready() or when your app modules are imported.
  warnings.warn(self.APPS_NOT_READY_WARNING_MSG, category=RuntimeWarning)
```